torch/csrc/jit/passes/onnx/unpack_quantized_weights.cpp

```diff
@@ -436,9 +521,54 @@ void unpackQuantizedWeightsHelper(
     eraseUnusedValuesFromMap(valsToParamsMap);
   }
 }
+
+static std::unordered_map<c10::ScalarType, c10::ScalarType, ScalarTypeHashFunction> qTypeToValTypeMap = {
+  {c10::ScalarType::QInt8, c10::ScalarType::Char},
+  {c10::ScalarType::QUInt8, c10::ScalarType::Byte},
+  {c10::ScalarType::QInt32, c10::ScalarType::Int},
+  {c10::ScalarType::QUInt4x2, c10::ScalarType::Byte},
+};
+
+// Unpack quantized tensor inputs into {value, scale, zero_point},
+// Then create a prim::TupleConstruct node based on these three values.
+void UnpackQuantizedTensorInputs(std::shared_ptr<Graph>& graph) {
+  bool search_quantize_tensors = true;
+  while (search_quantize_tensors) {
```

Comment:
I wonder how this could work. When we insertInput and eraseInput, `num_inputs` has been changed (+2 each time). Then the loop can not reach the end of the inputs.